src/rgb/resolvers.rs

```diff
@@ -1,40 +1,48 @@
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+use amplify::hex::ToHex;
+use bitcoin::Script;
+use bp::{LockTime, Outpoint, SeqNo, Tx, TxIn, TxOut, TxVer, Txid, VarIntArray, Witness};
+use rgb::{prelude::DeriveInfo, MiningStatus, Utxo};
+use rgbstd::{resolvers::ResolveHeight, validation::ResolveTx as ResolveCommiment};
 use std::{
-    collections::{BTreeMap, BTreeSet},
+    collections::{BTreeMap, BTreeSet, HashMap},
     convert::Infallible,
     str::FromStr,
 };
-
-use amplify::hex::ToHex;
-use bdk::blockchain::EsploraBlockchain;
-use bitcoin::Script;
-use bitcoin_hashes::hex::FromHex;
-use bp::{LockTime, Outpoint, Tx, TxIn, TxOut, Txid, VarIntArray};
-use futures::executor;
-use rgb::{
-    prelude::{DeriveInfo, MiningStatus},
-    Utxo,
-};
-use rgbstd::{resolvers::ResolveHeight, validation::ResolveTx as ResolveCommiment};
 use wallet::onchain::{ResolveTx, TxResolverError};
 
+#[derive(Default)]
 pub struct ExplorerResolver {
     pub explorer_url: String,
+    // Prefetch Data (wasm32)
+    pub utxos: BTreeSet<Utxo>,
+    pub next_utxo: String,
+    pub txs: HashMap<bitcoin::Txid, bitcoin::Transaction>,
+    pub bp_txs: HashMap<Txid, Tx>,
 }
```

Comment:
Very interesting... Very clever solution. I worry this might return false negatives if the transfer just took place, but we can work around that, yes?

Reply:
Yes, we can.

For now, I only add transactions to the hashmap if they actually exist in the blockchain/mempool, so I think it should work.

Reply:
That could work!